var/spack/repos/builtin/packages/cernlib/package.py

```diff
@@ -36,6 +39,10 @@ class Cernlib(CMakePackage):
     def patch(self):
         filter_file("crypto", "crypt", "packlib/CMakeLists.txt")
 
+    @when("@2023.08.14.0-free")
```

Comment:
This can be folded into the previous patch (outside of the GitHub context I can make suggestions on):
```suggestion
    @when("@:2023.08.14.0-free")
```
Typically, closed version ranges are used when a patch is reported upstream and in progress of being folded into the next release. If this isn't the case here, I'd remove the `when` entirely and apply the patch to all versions.